### DISCUSSION TITLE
AdGuard URL Tracking filter - Complete Mode

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBOL-home#98] [Q&A] [answered]
Question, by default I use uBlock Origin Lite in “basic mode” and configure it to “complete” on a per site basis. When I add additional annoyance filters or misc. filters, I notice the AdGuard URL tracking filter is grayed out. 

Does MV3 allow extended filters like AdGuard URL to work on “complete” configured sites, or for the filter to work, does the default setting of uBOL need to be configured to “optimized” or “complete”?

**Comment by MasterKia** [comment, marked as answer]
> for the AdGuard URL Tracking filter to work, does the default setting of uBOL need to be configured to “optimized” or “complete”?

Yes.

https://github.com/uBlockOrigin/uBOL-home/wiki/Frequently-asked-questions-(FAQ)#filtering-capabilities-which-cant-be-enforced-without-broad-readmodify-permissions:
> The following [modifier filter options](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#modifier-filters) can't be enforced without broad read/modify permissions, i.e. when default mode is _Basic_: ..., `removeparam=`, .... They will be enforced when default mode is either _Optimal_ or _Complete_.
>
> **For instance, this explains why the rule count of _AdGuard URL Tracking Protection_ is zero in _Basic_ mode, since the whole ruleset consists of `removeparam=` filters.**
